rfd/0011-database-access.md

```diff
@@ -0,0 +1,698 @@
+    aws:
+      # Optional AWS region RDS/Aurora database is running in.
+      region: "us-east-1"
+    # Static labels assigned to the database instance, used in RBAC.
+    static_labels:
+      env: "stage"
+    # Dynamic labels assigned to the database instance, used in RBAC.
+    dynamic_labels:
+    - name: "time"
+      command: ["date", "+%H:%M:%S"]
+      period: "1m"
+```
+
+When connecting Teleport database service to the cluster, users can either use
+a static join token (of type `db`) from auth server config, or generate a new
+join token for the service:
+
+```sh
+$ tctl tokens add --type=db
+```
+
+#### RDS/Aurora
+
+When configuring Teleport database service with RDS/Aurora databases, the
+following fields need to be set:
+
+```yaml
+db_service:
+  enabled: "yes"
+  databases:
+  - name: "postgres-rds"
+    protocol: "postgres"
+    uri: "postgres-rds.xxx.us-east-1.rds.amazonaws.com:5432"
+    ca_cert_file: "/opt/rds/rds-ca-2019-root.pem"
+    aws:
+      region: "us-east-1"
+```
+
+* `uri` is the endpoint for a particular database from RDS control panel.
+* `ca_cert_file` is an optional path to the [RDS root certificate](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html).
+* `aws.region` is the region the database is deployed in.
+
+If RDS root certificate is not provided explicitly, Teleport database service
+will attempt to download the correct version (based on the specified region
+the database is running in) automatically from AWS.
+
+AWS credentials will be initialized using default credential provider chain
+used by Go SDK which looks in the environment variables, then shared
+credentials file and then falls back to IAM role. Refer to [AWS documentation](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials)
+for more info.
+
+### Database instance
+
+#### Onprem
+
+To support mutual TLS auth, onprem database instance needs to be configured
+with Teleport's certificate authority and (optionally) cert/key pair issued
+by Teleport. The existing `tctl auth sign` command is used to produce these
+secrets:
+
+```sh
+$ tctl auth sign --format=db --host=localhost --out=server --ttl=8760h
+$ ls
+server.cas server.crt server.key
+```
+
+Note that the `--host` parameter should match the hostname of the endpoint
+the database will be connected at.
+
+The certificate is signed by Teleport host authority which is intended for
+machine-to-machine communication.
+
+The generated secrets are then used to configure the database. For example,
+for PostgreSQL:
+
+```sh
+$ cat postgresql.conf | grep ssl
+ssl = on
+ssl_ca_file = '/path/to/server.cas'
+ssl_cert_file = '/path/to/server.crt'
+ssl_key_file = '/path/to/server.key'
+```
+
+Database server may be configured with cert/key pair signed by a user's custom
+CA (for example, their organization's) in which case they would still need to
+supply Teleport's host CA so the server can authenticate Teleport client:
```

Comment:
```suggestion
supply Teleport's user CA so the server can authenticate Teleport client:
```

Reply:
It actually uses the host CA to sign database certificates as I felt this fits the "machine-to-machine" (or "dbservice-to-database") communication type, so the way it's phrased right now is correct. The "Teleport client" mentioned here refers to the "database service", not tsh.